pkg/server/controller/controller_test.go

```diff
@@ -0,0 +1,217 @@
+package controller
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/Lighty0410/ekadashi-server/pkg/crypto"
+	"github.com/Lighty0410/ekadashi-server/pkg/storage"
+	"github.com/stretchr/testify/require"
+)
+
+func TestController_RegisterUser(t *testing.T) {
+	mockService := &storage.ServiceMock{
+		AddUserFunc: func(in1 *storage.User) error {
+			if in1.Name == "username" {
+				return fmt.Errorf("duplicate key error collection")
+			}
+			if in1.Name == "somethingExceptable" {
+				return fmt.Errorf("cannot add user")
+			}
+			return nil
+		},
+	}
+	c := NewController(mockService)
+	tt := []struct {
+		casename      string
```

Comment:
name this field consistent through all you tests, either 'casename' or 'caseName'